File: Sample/sample_cs/Sound/Sound.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace sample_cs.Sound
{
	/// <summary>
	/// 音を鳴らすサンプル
	/// </summary>
	class Sound : ISample
	{
		public void Run()
		{
			// aceを初期化する。
			asd.Engine.Initialize("Sound", 640, 480, new asd.EngineOption());

			// ファイルを読み込む
			asd.SoundSource se1 = asd.Engine.Sound.CreateSoundSource("Data/Sound/se1.wav", true);
			asd.SoundSource bgm1 = asd.Engine.Sound.CreateSoundSource("Data/Sound/bgm1.ogg", false);

			// 再生する。
			int id_se1 = asd.Engine.Sound.Play(se1);
			int id_bgm1 = asd.Engine.Sound.Play(bgm1);

			// aceが進行可能かチェックする。
			while (asd.Engine.DoEvents())
			{
				// aceを更新する。
				asd.Engine.Update();

				// 音が再生終了してるか調べる。
				if(!asd.Engine.Sound.IsPlaying(id_bgm1))
				{
					break;
				}
			}

			// aceを終了する。
			asd.Engine.Terminate();
		}
	}
}

File: Sample/BasicSample/sample_cpp/Sound/Sound.cpp

#include <Altseed.h>

// Soundを用いて音を鳴らすサンプルです。
void Sound()
{
	// Altseedを初期化する。
	asd::Engine::Initialize(asd::ToAString("Sound").c_str(), 640, 480, asd::EngineOption());

	// 音ファイルを読み込む。
	std::shared_ptr<asd::SoundSource> se1 = asd::Engine::GetSound()->CreateSoundSource(asd::ToAString("Data/Sound/se1.wav").c_str(), true);
	std::shared_ptr<asd::SoundSource> bgm1 = asd::Engine::GetSound()->CreateSoundSource(asd::ToAString("Data/Sound/bgm1.ogg").c_str(), false);

	// 音を再生する。
	int id_se1 = asd::Engine::GetSound()->Play(se1);
	int id_bgm1 = asd::Engine::GetSound()->Play(bgm1);

	// Altseedのウインドウが閉じられていないか確認する。
	while (asd::Engine::DoEvents())
	{
		// Altseedを更新する。
		asd::Engine::Update();

		// 音が再生終了してるか調べる。
		if (!asd::Engine::GetSound()->GetIsPlaying(id_bgm1))
		{
			break;
		}
	}

	// Altseedの終了処理をする。
	asd::Engine::Terminate();

	return;
}
